docs/collecting-data/collecting-from-own-applications/java-tracker/previous-versions/java-tracker-v0-12/custom-tracking-using-schemas/index.md

```diff
@@ -38,7 +38,7 @@ The Java tracker does not yet provide the ability to automatically assign entiti
 
 The Java tracker provides the `SelfDescribingJson` class for custom events and entities. There is no in-built distinction between schemas used for events and those used for entities: they can be used interchangably.
 
-Your schemas must be accessible to your pipeline, within an [Iglu server](/docs/pipeline-components-and-applications/iglu/index.md). Tracked events containing self-describing JSON are validated against their schemas during the enrichment phase of the pipeline. If the data don't match the schema, the events end up in the Bad Rows storage instead of the data warehouse.
```

Comment:
This is the kind of sentences I want to avoid, as it implies that failed events don’t make it to the warehouse :)